### PR TITLE
SAAS-3131: Added Instance Exported Reports  Encrypted Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

File: collectors/aws/collector.js [hunk 1]
```diff
@@ -1045,6 +1045,11 @@ var postcalls = [
                 reliesOnService: 'connect',
                 reliesOnCall: 'listInstances',
                 override: true
+            },
+            listInstanceExportedReportsStorageConfigs: {
```
Comment (AkhtarAmir) [inline collectors/aws/collector.js:1049]: Can you change it to listInstanceExportedReportStorageConfigs

Comment (AkhtarAmir) [inline collectors/aws/collector.js:1049]: Can you change it to listInstanceExportedReportStorageConfigs

File: exports.js [hunk 2]
```diff
@@ -411,7 +411,9 @@ module.exports = {
       
         'networkMemberDataEncrypted'    : require(__dirname + '/plugins/aws/managedblockchain/networkMemberDataEncrypted.js'),
 
-        'instanceCallRecordingEncrypted': require(__dirname + '/plugins/aws/connect/instanceCallRecordingEncrypted.js')
+        'instanceCallRecordingEncrypted': require(__dirname + '/plugins/aws/connect/instanceCallRecordingEncrypted.js'),
+        'instanceExportedReportsEncrypted': require(__dirname + '/plugins/aws/connect/instanceExportedReportsEncrypted.js')
```
Comment (AkhtarAmir) [inline exports.js:415]: We need to shorten this name

File: exports.js [hunk 3]
```diff
@@ -411,7 +411,9 @@ module.exports = {
...
+
```
Comment (AkhtarAmir) [inline exports.js:416]: Remove extra line

File: plugins/aws/connect/instanceExportedReportsEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Connect Instance Exported Reports Encrypted',
+    category: 'Connect',
+    domain: 'Content Delivery',
+    description: 'Ensure that Amazon Connect instances have encryption enabled for exported reports being saved on S3.',
+    more_info: 'You can configure Amazon Connect instance to save exported reports to be saved on S3. When you save ' +
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceExportedReportsEncrypted.js:9]:
```suggestion
    more_info: 'You can configure Amazon Connect instance to save exported reports on S3. When you save ' +
```

File: plugins/aws/connect/instanceExportedReportsEncrypted.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        'such data on S3, enable encryption for the data and use a KMS key with desired encrypted level to meet regulatory compliance requirements within your organization.',
+    link: 'https://docs.aws.amazon.com/connect/latest/adminguide/encryption-at-rest.html',
+    recommended_action: 'Modify Connect instance data storage configuration and enable encryption for exported reports',
+    apis: ['Connect:listInstances', 'Connect:listInstanceExportedReportsStorageConfigs', 'KMS:listKeys', 'KMS:describeKey'],
+    settings: {
+        connect_exported_reports_encryption_level: {
+            name: 'Connect exported reports Encryption Level',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceExportedReportsEncrypted.js:16]: The initials should be capital
Connect Exported Reports Target Encryption Level

File: plugins/aws/connect/instanceExportedReportsEncrypted.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                        var currentEncryptionLevelString = helpers.ENCRYPTION_LEVELS[currentEncryptionLevel];
+
+                        if (currentEncryptionLevel >= desiredEncryptionLevel) {
+                            helpers.addResult(results, 0,
+                                `Connect instance is using ${currentEncryptionLevelString} for exported reports encryption\
+                                which is greater than or equal to the desired encryption level ${config.desiredEncryptionLevelString}`,
+                                region, resource);
+                        } else {
+                            helpers.addResult(results, 2,
+                                `Connect instance is using ${currentEncryptionLevelString} for exported reports encryption\
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceExportedReportsEncrypted.js:113]:
```suggestion
                                `Connect instance is using ${currentEncryptionLevelString} for exported reports encryption \
```

File: plugins/aws/connect/instanceExportedReportsEncrypted.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+                            continue;
+                        }
+    
+                        currentEncryptionLevel = helpers.getEncryptionLevel(describeKey.data.KeyMetadata, helpers.ENCRYPTION_LEVELS);
+
+                        var currentEncryptionLevelString = helpers.ENCRYPTION_LEVELS[currentEncryptionLevel];
+
+                        if (currentEncryptionLevel >= desiredEncryptionLevel) {
+                            helpers.addResult(results, 0,
+                                `Connect instance is using ${currentEncryptionLevelString} for exported reports encryption\
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceExportedReportsEncrypted.js:108]:
```suggestion
                                `Connect instance is using ${currentEncryptionLevelString} for exported reports encryption \
```

File: plugins/aws/connect/instanceExportedReportsEncrypted.spec.js [hunk 8]
```diff
@@ -0,0 +1,200 @@
+var expect = require('chai').expect;
+const instanceExportedReportsEncrypted = require('./instanceExportedReportsEncrypted');
+
+const listInstances = [
+    {
+        "Id": "9da26b8e-8f9e-4af2-8717-6f79913f2439",
+        "Arn": "arn:aws:connect:us-east-1:101363889637:instance/9da26b8e-8f9e-4af2-8717-6f79913f2439",
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceExportedReportsEncrypted.spec.js:7]: Change account id

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**